src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1948,14 +1948,12 @@ protected void EnsureNullabilityAnalysisPerformedIfNecessary()
 
             upgradeableLock.EnterWrite();
 
-            Debug.Assert(Root == bindableRoot);
-
             NullableWalker.SnapshotManager snapshotManager;
             var remappedSymbols = _parentRemappedSymbolsOpt;
             BoundNode boundRoot;
             Binder binder;
 
-            bind(bindableRoot, out binder, out boundRoot);
+            bind(Root, out binder, out boundRoot);
```

Comment:
>Root [](start = 17, length = 4)

Is there a requirement to bind Root in our case? Is this going to have an observable difference for our case? This feels fragile,  GetBindableSyntaxNode is designed to give back the node that can be bound on its own, other consumers are likely to be negatively impacted. I would like to better understand the reason for the adjustments in this function. If it is just due to the Assert, I would prefer us to remove, or adjust the assert instead.

Reply:
No, as the relevant semantics are already encoded in the binder.

---
In reply to: [444547175](https://github.com/dotnet/roslyn/pull/45334#discussion_r444547175) [](ancestors = 444547175)